### PR TITLE
Various Fixes for Sequence Navigation Stuff

### PR DESCRIPTION
**Pull request by darkruby501**
- migration to undo clobbering (done)
- fix layout shift (incomplete)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208673495664457) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      const updates = postsToUnset.map(post => ({
+        updateOne: {
+          filter: { _id: post._id },
+          update: {
+            $unset: {
+              canonicalPrevPostSlug: "",
+              canonicalNextPostSlug: "",
+              canonicalBookId: "",
+              canonicalCollectionSlug: "",
+              canonicalSequenceId: "",
```
Comment (b0b3rt) [inline packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:88]: Huh, surprised that this is the syntax for bulk unset operations.  (Thought it does match up to other uses.)

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      }
+
+      const postsInSequence = await getPostsInSequence(sequenceId);
+      const postsToUpdateInOrder = postsInSequence.filter(post => postIdsToUpdate.includes(post._id));
+
+      const updates = [];
+      for (let i = 0; i < postsToUpdateInOrder.length; i++) {
+        const currentPost = postsToUpdateInOrder[i];
+        const prevPost = i > 0 ? postsToUpdateInOrder[i - 1] : null;
+        const nextPost = i < postsToUpdateInOrder.length - 1 ? postsToUpdateInOrder[i + 1] : null;
```
Comment (b0b3rt) [inline packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:118]: I hope there aren't any off-by-one errors here - if you've tested this in dev, check the first, second, second-to-last, and last posts of a fixed sequence?

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        updates.push({
+          updateOne: {
+            filter: { _id: currentPost._id },
+            update: {
+              $set: {
+                canonicalPrevPostSlug: prevPost ? prevPost.slug : '',
+                canonicalNextPostSlug: nextPost ? nextPost.slug : '',
+                canonicalSequenceId: sequenceId,
```
Comment (b0b3rt) [inline packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:127]: Any chance these'll be missing their previous canonical book/collection IDs?

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+      if (updates.length > 0) {
+        await Posts.rawCollection().bulkWrite(updates, { ordered: false });
+        // eslint-disable-next-line no-console
+        console.log(`Updated canonical fields for ${updates.length} posts in sequence ${sequenceId}.`);
+      }
+    }
+  }
+});
+
+async function getPostsInSequence(sequenceId: string): Promise<DbPost[]> {
```
Comment (b0b3rt) [inline packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:143]: Don't we already have `sequenceGetAllPosts`?

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts (hunk 1 above)
Comment (darkruby501) [reply packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:88]: I questioned it too (as in questioned the model about the empty strings)

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts (hunk 3 above)
Comment (darkruby501) [reply packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:127]: These are all for user-made sequences, and users can't make collections (there are only 5) or books, so I don't think so?

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts (hunk 4 above)
Comment (darkruby501) [reply packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:143]: Ah, we do, but I've already tested it with the other one

File: packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts (hunk 2 above)
Comment (darkruby501) [reply packages/lesswrong/server/manualMigrations/2024-10-31-undoBestOfCollectionClobbering.ts:118]: I've tested first and second but haven't found last. This only fixes individual posts so I'd have to go find ones that happened to be last in a sequence, which is a pain to do.

**Review by Discordius — APPROVED** [review]

**Pull request merged**